packages/vehicle-rental-overlay/src/index.tsx

```diff
@@ -192,44 +179,40 @@ const VehicleRentalOverlay = ({
         </Source>
       )}
       {zoom >= DETAILED_MARKER_CUTOFF &&
-        stations
-          .filter(station =>
-            checkIfPositionInViewport(bounds, station.y, station.x)
```

Comment:
Was this filter causing rendering issues?

Reply:
It was. But it's not needed anymore. Vector tiles are much more performant than the raster ones

Reply:
How about rendering extra HTML that won't get shown except for the small portion of bike stations within the viewport?

Reply:
Why would we do this?

Reply:
Removing that filter results in more markup sent to the browser that is not directly visible (actually more markup sent to maplibre for rendering). Was that the intent? Perhaps one case for removing the filter is to prevent re-rendering when slightly panning the map.

Reply:
If we render/unrender markup using react we are now relying on react to do a map refresh instead of letting maplibre handle it. Maplibre does a good job at deciding whether to render something or not.

Reply:
Good deal.